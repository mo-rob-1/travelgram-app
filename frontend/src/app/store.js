import { configureStore } from "@reduxjs/toolkit";
import authReducer from "../features/auth/authSlice";
import imageReducer from "../features/images/imageSlice";

export const store = configureStore({
  reducer: {
    auth: authReducer,
    images: imageReducer,
  },
});
