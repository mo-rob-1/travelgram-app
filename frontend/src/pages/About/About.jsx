import { Section, Span } from "./About.styled";
import { Link } from "react-router-dom";
import { Helmet } from "react-helmet";

function About() {
  return (
    <Section>
      <Helmet>
        <title>About | travelGram App</title>
      </Helmet>
      <h1>
        About <Span>travel</Span>Gram
      </h1>
      <h3 style={{ marginBottom: "1rem" }}>
        Welcome to <Span>travel</Span>Gram
      </h3>
      <p>travelGram is an app that allows you to upload and share your travel images.</p>
      <p style={{ marginTop: "1rem", marginBottom: "1rem" }}>This app is perfect for:</p>
      <ul>
        <li>Sharing your travel images with your friends and family.</li>
        <li>Professional and amateur photographers wanting to share their portfolio.</li>
        <li>Backpackers wanting to share their journey.</li>
        <li>Gap year students wanting to share the places they’ve been to with their friends and colleagues.</li>
        <li>
          You – Have you been somewhere or went traveling and want to share the pictures you’ve taken? If yes, then this
          app is for you.{" "}
          <Link to="/register" style={{ color: "#222", fontWeight: "bold" }}>
            Register
          </Link>{" "}
          now to get started.
        </li>
      </ul>
    </Section>
  );
}

export default About;
