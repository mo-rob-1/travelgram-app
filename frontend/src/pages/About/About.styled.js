import styled from "styled-components";

export const Section = styled.section`
  margin-top: 1rem;
  padding-left: 1rem;

  @media (min-width: 1024px) {
    margin-top: 3rem;
  }
`;

export const Span = styled.span`
  font-weight: 400;
`;
