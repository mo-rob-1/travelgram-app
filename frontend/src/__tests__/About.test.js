import { render } from "@testing-library/react";
import { createMemoryHistory } from "history";
import { Provider } from "react-redux";
import { store } from "../app/store";
import { BrowserRouter as Router } from "react-router-dom";
import About from "../pages/About/About";

describe("About", () => {
  it("renders the about page", () => {
    const history = createMemoryHistory();
    const { asFragment } = render(
      <Provider store={store} history={history}>
        <Router>
          <About />
        </Router>
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
