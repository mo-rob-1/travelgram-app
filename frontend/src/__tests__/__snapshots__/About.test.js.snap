// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About renders the about page 1`] = `
<DocumentFragment>
  <section
    class="sc-bczRLJ boIiCX"
  >
    <h1>
      About 
      <span
        class="sc-gsnTZi gQtzFr"
      >
        travel
      </span>
      Gram
    </h1>
    <h3
      style="margin-bottom: 1rem;"
    >
      Welcome to 
      <span
        class="sc-gsnTZi gQtzFr"
      >
        travel
      </span>
      Gram
    </h3>
    <p>
      travelGram is an app that allows you to upload and share your travel images.
    </p>
    <p
      style="margin-top: 1rem; margin-bottom: 1rem;"
    >
      This app is perfect for:
    </p>
    <ul>
      <li>
        Sharing your travel images with your friends and family.
      </li>
      <li>
        Professional and amateur photographers wanting to share their portfolio.
      </li>
      <li>
        Backpackers wanting to share their journey.
      </li>
      <li>
        Gap year students wanting to share the places they’ve been to with their friends and colleagues.
      </li>
      <li>
        You – Have you been somewhere or went traveling and want to share the pictures you’ve taken? If yes, then this app is for you. 
        <a
          href="/register"
          style="color: rgb(34, 34, 34);"
        >
          Register
        </a>
         now to get started.
      </li>
    </ul>
  </section>
</DocumentFragment>
`;
